File: APP_ENGINE/BlueCompass/src/main/java/com/singhtwenty2/OceanVista/data/repository/BeachRepository.java
package com.singhtwenty2.OceanVista.data.repository;

import com.singhtwenty2.OceanVista.data.model.entity.Beach;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface BeachRepository extends JpaRepository<Beach, Long> {
}

File: NATIVE_ANDROID/app/src/main/java/com/singhtwenty2/oceanvista/feature_home/domain/repository/BeachRepository.kt
package com.singhtwenty2.oceanvista.feature_home.domain.repository

import com.singhtwenty2.oceanvista.feature_home.domain.model.dto.response.Beach
import com.singhtwenty2.oceanvista.feature_home.util.HomeApiResponseHandler

interface BeachRepository {
    suspend fun getBeaches(): HomeApiResponseHandler<List<Beach>>
}